lib/hm_crypto/container.ex

```diff
@@ -117,6 +119,18 @@ defmodule HmCrypto.Container do
     <<0x00>> <> add_paddings(serial_number <> nonce <> <<0x01>> <> data) <> <<0xFF>>
   end
 
+  @spec enclose(
+          command,
+          serial_number,
+          serial_number,
+          Crypto.private_key(),
+          HmCrypto.AccessCertificate.access_certificate_binary() | HmCrypto.Crypto.public_key(),
+          nonce,
+          binary,
+          ContentType.t(),
+          :v2
+        ) :: secure_command
+
   def enclose(
```

Comment:
Credo complains about having too many parameters here. 

```
Function takes too many parameters (arity is 9, max is 8).
┃       lib/hm_crypto/container.ex:134:7 #(HmCrypto.Container.enclose)
```

We can change the credo config for now, but maybe it's worth it to consider in the future how to make this better.

One option could be to use a creation pattern similar to `Plug.Conn`, for example:

```
Container.new()
|> put_command(...)
|> put_sender_serial_number(...)
...
|> enclose()
```

Reply:
At first looks like a nice approach but considering that this function call only happens in one place compare to plug that this data pushes to the struct while plug is pass through.

I'll change the function to accept Container struct.